bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -113,7 +113,7 @@ public void removeSubscription(String subscriptionId) {
             // this was the only subscription listening on this page, so we can dispose the listener
             PageChangeListener listener = pageChangeListeners.get(sitemapPage);
             if (listener != null) {
-                pageChangeListeners.remove(listener);
+                pageChangeListeners.remove(sitemapPage);
```

Comment:
As we want to remove the entry from the map in every case you could simplify use:
```java
final PageChangeListener listener = pageChangeListeners.remove(sitemapPage);
if (listener != null) {
    listener.dispose();
}
```